vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp

```diff
@@ -160,10 +169,14 @@ void MapBufferBuilder::putMapBuffer(Key key, MapBuffer &map) {
 }
 
 MapBuffer MapBufferBuilder::build() {
+  react_native_assert(
+      (keyValues_ != nullptr) &&
+      "Error when building mapbuffer with invalid datastructures.");
+
   // Create buffer: [header] + [key, values] + [dynamic data]
-  int bufferSize = keyValuesOffset_ + dynamicDataOffset_;
+  int32_t bufferSize = keyValuesOffset_ + dynamicDataOffset_;
 
-  _header.bufferSize = static_cast<uint16_t>(bufferSize);
```

Comment:
Removing this fails my local build (I include the updated .h file too).
@NickGerleman Have you tried it locally?

Reply:
It seemed to succeed locally. There was a recent upstream change that I think changed sizes. Do you see anything different if you do a clean build?

Reply:
I made an update to the core PR and found this to be needed. I guess I must have done something wrong, since the CI passes here. 
Still, applying this change to a local master from a few days ago fails for me.

Reply:
This integration itself pulls in MapBuffer changes to use int32_t in more places. So I think the removal may only work now.

Reply:
This pulls in all the latest changes to MapBuffer from core. If this works, then we can actually remove the forked file, since it's now no different from core, I think.

Reply:
There's one last cast still, from size_t to int32_t. The uint16_t ones should be gone though.

Reply:
Double-checked and the int32_t cast was added on Apr 15, so we could have deleted this file in this PR.